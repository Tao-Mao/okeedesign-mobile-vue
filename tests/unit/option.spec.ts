import { mount } from '../index';
import { Checker as CheckerNative, CheckerItem as CheckerItemNative } from '@src';
const Checker = CheckerNative as any;
const CheckerItem = CheckerItemNative as any;

describe('Checker', () => {
  test('render base', () => {
    const wrapper = mount({
      template: `
        <o-checker v-model="checker">
          <o-row gutter="10">
            <o-col span="8">
              <o-checker-item label="0" :value="0" />
            </o-col>
            <o-col span="8">
              <o-checker-item label="1" value="1" />
            </o-col>
            <o-col span="8">
              <o-checker-item value="2">2</o-checker-item>
            </o-col>
          </o-row>
        </o-checker>
      `,
      components: {
        'o-checker': Checker,
        'o-checker-item': CheckerItem,
      },
      data() {
        return {
          checker: [0, '2'],
        };
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('inputType radio', () => {
    const wrapper = mount({
      template: `
        <o-checker v-model="checker">
          <o-checker-item label="0" :value="0" />
          <o-checker-item label="1" value="1" />
          <o-checker-item value="2">2</o-checker-item>
        </o-checker>
      `,

      components: {
        'o-checker': Checker,
        'o-checker-item': CheckerItem,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Event triggered when the binding value changes', async () => {
    const onClick = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount({
      template: `
        <o-checker 
          v-model="checker"
          @click="onClick"
          @change="onChange">
          <o-checker-item label="0" :value="0" />
          <o-checker-item label="1" value="1" />
          <o-checker-item value="2">2</o-checker-item>
        </o-checker>
      `,
      data() {
        return {
          checker: ['2'],
        };
      },
      methods: {
        onClick,
        onChange,
      },
      components: {
        'o-checker': Checker,
        'o-checker-item': CheckerItem,
      },
    });
    const items = wrapper.findAll('.o-checker-item');
    await items.at(0).trigger('click');

    expect(onClick).toBeCalledTimes(1);
    expect(onChange).toBeCalledTimes(1);
  });
});
