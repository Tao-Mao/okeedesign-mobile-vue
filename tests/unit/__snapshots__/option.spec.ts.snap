// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checker inputType radio 1`] = `
<div class="o-checker"><label class="o-checker-item o-checker-item--default o-checker-item--normal"><input type="checkbox" name="" value="0">
    <div class="o-checker-item__label">0</div>
  </label> <label class="o-checker-item o-checker-item--default o-checker-item--normal"><input type="checkbox" name="" value="1">
    <div class="o-checker-item__label">1</div>
  </label> <label class="o-checker-item o-checker-item--default o-checker-item--normal"><input type="checkbox" name="" value="2">2</label></div>
`;

exports[`Checker render base 1`] = `
<div class="o-checker">
  <div class="o-row" style="margin-left: -5px; margin-right: -5px;">
    <div class="o-col o-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="o-checker-item o-checker-item--default o-checker-item--normal o-checker-item--active"><input type="checkbox" name="" value="0">
        <div class="o-checker-item__label">0</div><i class="o-checker-item__icon" style="width: .8rem; height: .8rem;">test-file-stub</i>
      </label></div>
    <div class="o-col o-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="o-checker-item o-checker-item--default o-checker-item--normal"><input type="checkbox" name="" value="1">
        <div class="o-checker-item__label">1</div>
      </label></div>
    <div class="o-col o-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="o-checker-item o-checker-item--default o-checker-item--normal o-checker-item--active"><input type="checkbox" name="" value="2">2<i class="o-checker-item__icon" style="width: .8rem; height: .8rem;">test-file-stub</i></label></div>
  </div>
</div>
`;
